jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/ApacheUpnpServiceConfiguration.java

```diff
@@ -48,6 +49,8 @@ public StreamClient createStreamClient() {
     @SuppressWarnings("rawtypes")
     @Override
     public StreamServer createStreamServer(NetworkAddressFactory networkAddressFactory) {
-        return new StreamServerImpl(new StreamServerConfigurationImpl(networkAddressFactory.getStreamListenPort()));
+        int listenPort = SettingsService.getDefaultUPnPPort();
+        return new StreamServerImpl(new StreamServerConfigurationImpl(
+                listenPort < 1 ? networkAddressFactory.getStreamListenPort() : listenPort));
     }
```

Comment:
Jpsonic's UPnP port defaults to -1. (Represents unspecified). If the UPnP port number specified by the user is less than 1, Jpsonic's UPnP port determination has the same specifications as Subsonic. In other words, Auto detect is used.

If the value is above 1, then use that port value to launch UPnP.

Reply:
This specification is spec of Jpsonic, but the fix of https://github.com/tesshucom/jpsonic/commit/24d54fd36d5f89e93cc0ade31d14b9959da7cb73 was not enough and caused a bug.